lib/client.js

```diff
@@ -51,19 +52,8 @@ exports.boot = function boot(options, callback) {
       throw new Error('asking to use react router for rendering, but no routes are provided');
     }
 
-    // seems pointless to do this.
-    options.location = options.location || Router.HistoryLocation;
-
-    // create and run the router
-    var router = Router.create(options);
-
-    router.run(function onRouterRun(Component) {
-      // create a component instance
-      var componentInstance = React.createElement(Component, props);
-
-      // finally, render the component instance into the document
-      ReactDOM.render(componentInstance, _document);
-    });
+    var routerComponent = React.createElement(Router.Router, { routes: options.routes, history: history.createHistory() });
+    ReactDOM.render(routerComponent, _document);
```

Comment:
This change was missed by @chunkiat82 earlier. [test/client.js](https://github.com/paypal/react-engine/blob/v3.x/test/client.js) does not test this, only single components. Should raise an issue to add a test.